exist-core/src/main/java/org/exist/xquery/functions/util/Base64Functions.java

```diff
@@ -42,76 +42,80 @@
-        ),
-		
-         new FunctionSignature(
-            new QName( "base64-decode", UtilModule.NAMESPACE_URI, UtilModule.PREFIX ),
-            "Decode the given Base64 encoded string back to clear text",
-            new SequenceType[] {
-                new FunctionParameterSequenceType( "string", Type.STRING, Cardinality.ZERO_OR_ONE, "The Base64 string to be decoded" )
-            },
-            new FunctionReturnSequenceType( Type.STRING, Cardinality.ZERO_OR_ONE, "the decoded output" )
-        )
+public class Base64Functions extends BasicFunction {
+    protected static final Logger logger = LogManager.getLogger(Base64Functions.class);
+
+    public final static FunctionSignature signatures[] = {
+            new FunctionSignature(
+                    new QName("base64-encode", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Encodes the given string as Base64",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("string", Type.STRING, Cardinality.ZERO_OR_ONE, "The string to be Base64 encoded")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE, "the Base64 encoded output, with trailing newlines trimmed")
+            ),
+
+            new FunctionSignature(
+                    new QName("base64-encode", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Encodes the given string as Base64",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("string", Type.STRING, Cardinality.ZERO_OR_ONE, "The string to be Base64 encoded"),
+                            new FunctionParameterSequenceType("trim", Type.BOOLEAN, Cardinality.EXACTLY_ONE, "Trim trailing newlines?")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE, "the Base64 encoded output"),
+                    "This function is deprecated. The output does not need to be trimmed, please use util:base64-encode#1 instead."
+            ),
+
+            new FunctionSignature(
+                    new QName("base64-decode", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Decode the given Base64 encoded string back to clear text",
+                    new SequenceType[]{
+                            new FunctionParameterSequenceType("string", Type.STRING, Cardinality.ZERO_OR_ONE, "The Base64 string to be decoded")
+                    },
+                    new FunctionReturnSequenceType(Type.STRING, Cardinality.ZERO_OR_ONE, "the decoded output")
+            ),
+
+            new FunctionSignature(
+                    new QName("base64-encode-url-safe", UtilModule.NAMESPACE_URI, UtilModule.PREFIX),
+                    "Encodes the given string as Base64, url-safe. No padding and use - and _ instead of + and /.",
```

Comment:
The URL Safe alphabet used by Apache Commons Codec is only one of several possibilities (https://en.wikipedia.org/wiki/Base64#Variants_summary_table), with each being incompatible with each other. Are we certain this URL safe alphabet is the "defacto standard" which is also used for JWT etc?

Looking into the Commons Codec code, it's URL Safe variant appears to be compliant with RFC 4648. Do you agree? If so we should likely write the description to be more accurate like:

```
Encodes the given string into a Base64 alphabet as defined by RFC 4648 §5 Base 64 Encoding with URL and Filename Safe Alphabet.
```

Reply:
Well, I referenced RFC 4648 - this IS the de-facto standard.

Reply:
Explaining what it actually does in contrast to the default behaviour helps me - as a developer more than some abstract reference to an RFC.

Reply:
`(see RFC 4648 §5)` was added to description